### PR TITLE
Add json util to unmarshal numbers into int/float

### PR DESCRIPTION
**Pull request by hueifeng**
# Description

Details are as follows: https://github.com/dapr/dapr/pull/4069

## Issue reference

Please reference the issue this PR will close: #3837 #4026  

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by artursouza** [comment]
E2E tests are not passing on this PR. Moving to 1.7

**Comment by yaron2** [comment]
@hueifeng please see that e2e tests are failing. Need to understand why and fix.

**Comment by yaron2** [comment]
@hueifeng see linter failures. I recommend using an editor like VS code with the Go extension to make sure all the linter requirements are in place.

**Comment by yaron2** [comment]
@hueifeng linter and tests issues..

**Comment by hueifeng** [comment]
@yaron2 It looks like it's working.
image

**Comment by yaron2** [comment]
> @yaron2 It looks like it's working. image

Search for "Check failure" here: https://github.com/dapr/dapr/pull/4093/files. it'll show you exactly which files need what formatter.

**Comment by yaron2** [comment]
@hueifeng please run the following:

(Make sure you're using Go.17.x)

```
go install golang.org/x/tools/cmd/goimports@latest

goimports -w -local ./pkg/channel/testing/grpc_channel_server_mock.go
goimports -w -local ./pkg/injector/injector_test.go
goimports -w -local ./tests/apps/pubsub-subscriber_grpc/app.go

go install mvdan.cc/gofumpt@latest

gofumpt -w ./pkg/grpc/api.go
gofumpt -w ./pkg/grpc/api.go
gofumpt -w ./pkg/http/api.go
```

**Comment by yaron2** [comment]
@hueifeng we are almost there! down to 3 failures from 6.

Please run the following:

```
goimports -w -local ./pkg/operator/api/api.go
goimports -w -local ./pkg/components/kubernetes_loader_test.go
goimports -w -local ./pkg/expr/expr.go
```

**Comment by artursouza** [comment]
Moving this to 1.7 since we are cutting RC1 for 1.6.0. I can still cherry-pick it to 1.6 if this is merged prior to the release.

**Comment by hueifeng** [comment]
There is a conflict，I only have only read only permissions, there is no write permission.

**Comment by daixiang0** [comment]
@hueifeng you can do it locally then `git push`.

File: pkg/http/api.go [hunk 1]
```diff
@@ -1335,7 +1335,7 @@ func (a *api) onGetMetadata(reqCtx *fasthttp.RequestCtx) {
 		RegisteredComponents: registeredComponents,
 	}
 
-	mtdBytes, err := daprjson.Marshal(mtd)
+	mtdBytes, err := jsoniter.Marshal(mtd)
```
Comment (yaron2) [inline pkg/http/api.go:1338]: we need to remove `jsoniter` here

**Review by yaron2 — CHANGES_REQUESTED** [review]

File: pkg/http/api.go [hunk 2]
```diff
@@ -1713,7 +1714,7 @@ func (a *api) onQueryState(reqCtx *fasthttp.RequestCtx) {
 		qresp.Results[i].Data = jsoniter.RawMessage(resp.Results[i].Data)
```
Comment (Taction) [inline pkg/http/api.go:1714]: please replace `jsoniter.RawMessage` with `json.RawMessage`. So we can completely remove the package `github.com/json-iterator/go`.

**Pull request closed without merging**